pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -354,6 +367,9 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 	// Apply parameter substitution from the PipelineRun
 	pipelineSpec = resources.ApplyParameters(pipelineSpec, pr)
 
+	// pipelineState holds a list of pipeline tasks after resolving conditions and pipeline resources
+	// pipelineState also holds a taskRun for each pipeline task after the taskRun is created
+	// pipelineState is instantiated and updated on every reconcile cycle
```

Comment:
not something to necessarily solve in this pr (and also not a reason to remove the comment!), but if a variable needs a comment like this to explain what it's doing, it's probably trying to do more things than one variable should do

Reply:
:( hoping it will be helpful for fresh eyes ...